servers/rendering/rendering_device.cpp

```diff
@@ -707,6 +708,29 @@ RID RenderingDevice::texture_create(const TextureFormat &p_format, const Texture
 	format.height = format.texture_type != TEXTURE_TYPE_1D && format.texture_type != TEXTURE_TYPE_1D_ARRAY ? format.height : 1;
 	format.depth = format.texture_type == TEXTURE_TYPE_3D ? format.depth : 1;
 
+	uint64_t size_max = 0;
+	switch (format.texture_type) {
+		case TEXTURE_TYPE_1D:
+		case TEXTURE_TYPE_1D_ARRAY:
+			size_max = driver->limit_get(LIMIT_MAX_TEXTURE_SIZE_1D);
+			break;
+		case TEXTURE_TYPE_2D:
+		case TEXTURE_TYPE_2D_ARRAY:
+			size_max = driver->limit_get(LIMIT_MAX_TEXTURE_SIZE_2D);
+			break;
+		case TEXTURE_TYPE_CUBE:
+		case TEXTURE_TYPE_CUBE_ARRAY:
+			size_max = driver->limit_get(LIMIT_MAX_TEXTURE_SIZE_CUBE);
+			break;
+		case TEXTURE_TYPE_3D:
+			size_max = driver->limit_get(LIMIT_MAX_TEXTURE_SIZE_3D);
+			break;
+		case TEXTURE_TYPE_MAX:
+			//Make the compiler happy
+			break;
```

Comment:
I'd replace this with a default case.
```suggestion
		default:
			size_max = 0
			break;
```
I'm not sure if `0` by default is ideal by the way.

Reply:
I avoided using default because I wanted this code to fail to build if the enum was extended, rather than silently create a potential bug. `0` is more meant as a placeholder than a default; `case TEXTURE_TYPE_MAX` should actually be dead code because of the `ERR_FAIL_INDEX_V(format.texture_type, RDD::TEXTURE_TYPE_MAX, RID())` further up the function.